src/search/SearchMessagesScreen.js

```diff
@@ -131,6 +149,35 @@ class SearchMessagesScreenInner extends PureComponent<Props, State> {
     this.handleQuerySubmit(e);
   };
 
+  fetchOlder = async () => {
+    if (!this.props.loading && !this.state.foundOldest && !this.state.isFetching) {
+      this.setState({ isFetching: true });
+      try {
+        // Not sure why we require a double await here.
+        const { messages: fetchedMessages, foundOldest } = await await this.props.dispatch(
+          fetchMessages({
+            anchor: this.state.messages ? this.state.messages[0].id : LAST_MESSAGE_ANCHOR,
+            narrow: SEARCH_NARROW(this.state.query),
+            numBefore: config.messagesPerRequest,
+            numAfter: 0,
+          }),
+        );
+        // Last fetched message === first current message, hence remove it
+        fetchedMessages.pop();
```

Comment:
Not using `LAST_MESSAGE_ANCHOR` here anymore. I have added an invariant call here, since flow was complaining when I removed `LAST_MESSAGE_ANCHOR`, given that we always expect `this.state.messages` to be not null. Let me know if that's not a correct assumption.

Reply:
Sounds good.

I *think* it will always be non-null, but I'd have to stare at this concurrency logic harder to be certain there isn't a case where that could fail. Above I'd said:

> this callback should only get invoked when there are already some messages in the first place, and if there somehow aren't, it can just return without doing anything. That's what the old `fetchOlder` for `ChatScreen` does, after all.

which would mean to add that check to the early return at the top of this function. But in this case there's no practical difference to the user between this function returning early and throwing — its call site, the `props.fetchOlder()` in `handleOutboundEvents`, doesn't await it so won't know the difference. The only difference is that throwing will cause a Sentry event to be logged (because the Promise rejection will go unhandled.) And that seems for the best given that we believe the case where it's null here shouldn't be possible.